sklearn/datasets/_samples_generator.py

```diff
@@ -621,8 +624,16 @@ def make_circles(n_samples=100, shuffle=True, noise=None, random_state=None,
     if factor >= 1 or factor < 0:
         raise ValueError("'factor' has to be between 0 and 1.")
 
-    n_samples_out = n_samples // 2
-    n_samples_in = n_samples - n_samples_out
+    if isinstance(n_samples, (int, np.integer)):
```

Comment:
Should we create a helper function for this?

Reply:
I think it is straightforward. I searched whole project and I haven't found any similar usages where it could be reused. But sure, I can create a function: 

```python
def is_int(x):
    """Check x is an int.

    x can be a built-in int or any numpy int.

    Parameters
    ----------
    x : variable to check
    """
    return isinstance(x, (int, np.integer))

```
Where should it go, `utils.validation`?

Reply:
please use isinstance(x, numbers.Integral), see #14004